core/src/main/java/com/microsoft/applicationinsights/internal/processor/PageViewTelemetryFilter.java

```diff
@@ -0,0 +1,140 @@
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.internal.processor;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.microsoft.applicationinsights.agent.internal.common.StringUtils;
+import com.microsoft.applicationinsights.extensibility.TelemetryProcessor;
+import com.microsoft.applicationinsights.internal.annotation.BuiltInProcessor;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.telemetry.PageViewTelemetry;
+import com.microsoft.applicationinsights.telemetry.Telemetry;
+
+/**
+ * The class can filter out PageViewTelemetries that
+ *   have a duration which is less than a predefined value
+ *   have URLs that has parts that are not needed, based on configuration
+ *   have unneeded page names that were predefined in configuration
+ *
+ *  Illegal value will prevent from the filter from being used.
+ *
+ * Created by gupele on 7/26/2016.
+ */
+@BuiltInProcessor("PageViewTelemetryFilter")
+public final class PageViewTelemetryFilter implements TelemetryProcessor {
+
+    private long durationThresholdInMS = 0l;
+    private final Set<String> notNeededUrls = new HashSet<String>();
+    private final Set<String> notNeededNames = new HashSet<String>();
+
+    public PageViewTelemetryFilter() {
+    }
+
+    @Override
+    public boolean process(Telemetry telemetry) {
+        if (telemetry == null) {
+            return true;
+        }
+
+        if (!(telemetry instanceof PageViewTelemetry)) {
+            return true;
+        }
+
+        PageViewTelemetry asPVT = (PageViewTelemetry)telemetry;
+        URI uri = asPVT.getUri();
+        if (uri == null) {
+            return true;
+        } else {
+            String uriPath = uri.toString();
+            for (String notNeededUri : notNeededUrls) {
+                if (uriPath.contains(notNeededUri)) {
+                    return false;
+                }
+            }
+        }
+
+        if (notNeededNames.contains(asPVT.getName())) {
+            return false;
+        }
+
+        long pvtDuration = asPVT.getDuration();
```

Comment:
May be worth to do the getDuration validation first. It's more lightweight than the URL/Name scan and if the duration is below the threshold, we can avoid the lengthier string.contains operations